### DISCUSSION TITLE
Is there a way to pass manual parameters into tool calls

### DISCUSSION THREAD

**Discussion by Igrium** [sashirestela/simple-openai#232] [Q&A]
From what I can tell, tool calls are implemented via classes which are instantiated via reflection, which means I can't access any data about the context from inside the tool call.

Is there a clean way to pass parameters into a tool call outside of what the LLM gives it? (ex. state info about the current session)

**Comment by sashirestela** [comment]
Hi @Igrium thanks for using simple-openai. Regarding your question, I'm afraid not yet. I think this is related to what other users were asking for: #200. I don't have much free time to work on it soon, but if someone can prepare a PR, that would be great.
